components/dashboard/src/components/AuthorizeGit.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { AuthProviderInfo } from "@gitpod/gitpod-protocol";
+import { FC, useContext } from "react";
+import { Link } from "react-router-dom";
+import { useAuthProviders } from "../data/auth-providers/auth-provider-query";
+import { useCurrentOrg } from "../data/organizations/orgs-query";
+import { openAuthorizeWindow } from "../provider-utils";
+import { getGitpodService } from "../service/service";
+import { UserContext, useCurrentUser } from "../user-context";
+import { Button } from "./Button";
+import { Heading2, Subheading } from "./typography/headings";
+
+export function useNeedsGitAuthorization() {
+    const authProviders = useAuthProviders();
+    const user = useCurrentUser();
+    if (!user || !authProviders.data) {
+        return false;
+    }
+    return !authProviders.data.some((ap) => user.identities.some((i) => ap.authProviderId === i.authProviderId));
+}
+
+export function AuthorizeGit() {
+    const { setUser } = useContext(UserContext);
+    const org = useCurrentOrg();
+    const authProviders = useAuthProviders();
+    const updateUser = () => {
+        getGitpodService().server.getLoggedInUser().then(setUser);
+    };
+
+    const connect = (ap: AuthProviderInfo) => {
+        openAuthorizeWindow({
+            host: ap.host,
+            scopes: ap.requirements?.default,
+            overrideScopes: true,
+            onSuccess: updateUser,
+        });
+    };
+
+    if (authProviders.data === undefined) {
+        return <></>;
+    }
+
+    const verifiedProviders = authProviders.data.filter((ap) => ap.verified);
+
+    return (
+        <div className="w-full text-center bg-gray-100 dark:bg-gray-800 p-4 m-4 rounded-lg py-10">
```

Comment:
**suggestion**: Re-posting here for visibility from a [relevant comment](https://linear.app/gitpod/issue/WEB-246#comment-7e72d8f5) in WEB-246.

| No integrations, no permissions to add | No integrations, with permissions to add |
|-|-|
| <img width="1440" alt="Auth-1" src="https://user-images.githubusercontent.com/120486/236513778-d30f28f3-f568-4b46-ba01-c83010814b58.png"> | <img width="1440" alt="Auth-2" src="https://user-images.githubusercontent.com/120486/236513796-855e5bb4-57ad-49a5-8e90-28c51ea9fa0a.png"> |

| With (1) integration, without authorization | With multiple integrations, without authorization |
|-|-|
| <img width="1440" alt="Auth" src="https://user-images.githubusercontent.com/120486/236513895-5f01e574-d219-496c-a630-e2c613f213f5.png"> | <img width="1440" alt="Auth-3" src="https://user-images.githubusercontent.com/120486/236513902-81a02a66-ef4d-4084-8978-8b8bffcb59c8.png"> |